client/dashboard/me/notifications-extras/test/index.test.tsx

```diff
@@ -0,0 +1,366 @@
+		};
+
+		mockGetNotificationSettingsApi( customSettings );
+
+		dashboardRender(
+			<>
+				<Snackbars />
+				<NotificationsExtras />
+			</>
+		);
+
+		await waitFor( () => {
+			// WordPress.com section - first Suggestions toggle should be checked
+			const wpcomSuggestions = screen.getAllByLabelText( 'Suggestions' )[ 0 ];
+			expect( wpcomSuggestions ).toBeChecked();
+		} );
+
+		// WordPress.com toggles - verify enabled ones are checked
+		const wpcomSuggestions = screen.getAllByLabelText( 'Suggestions' )[ 0 ];
+		const wpcomResearch = screen.getAllByLabelText( 'Research' )[ 0 ];
+		expect( wpcomSuggestions ).toBeChecked();
+		expect( wpcomResearch ).toBeChecked();
+
+		// WordPress.com toggles - verify disabled ones are unchecked
+		expect( screen.getByLabelText( 'Community' ) ).not.toBeChecked();
+		expect( screen.getAllByLabelText( 'Promotions' )[ 0 ] ).not.toBeChecked();
+		expect( screen.getAllByLabelText( 'Newsletter' )[ 0 ] ).not.toBeChecked();
+		expect( screen.getByLabelText( 'Digests' ) ).not.toBeChecked();
+		expect( screen.getAllByLabelText( 'Reports' )[ 0 ] ).not.toBeChecked();
+		expect( screen.getByLabelText( 'Developer Newsletter' ) ).not.toBeChecked();
+		expect( screen.getByLabelText( 'Scheduled updates' ) ).not.toBeChecked();
+
+		// Jetpack toggles - verify enabled ones are checked
+		const jetpackSuggestions = screen.getAllByLabelText( 'Suggestions' )[ 1 ];
+		const jetpackNewsletter = screen.getAllByLabelText( 'Newsletter' )[ 1 ];
+		expect( jetpackSuggestions ).toBeChecked();
+		expect( jetpackNewsletter ).toBeChecked();
+
+		// Jetpack toggles - verify disabled ones are unchecked
+		expect( screen.getAllByLabelText( 'Research' )[ 1 ] ).not.toBeChecked();
+		expect( screen.getAllByLabelText( 'Promotions' )[ 1 ] ).not.toBeChecked();
+		expect( screen.getAllByLabelText( 'Reports' )[ 1 ] ).not.toBeChecked();
+	} );
+
+	it( 'shows "Subscribe to all" when no settings are enabled', async () => {
+		// Start with all settings false
+		mockGetNotificationSettingsApi();
+
+		dashboardRender(
+			<>
+				<Snackbars />
+				<NotificationsExtras />
+			</>
+		);
+
+		// Initially, all settings are false, so both sections should show "Subscribe to all"
+		await waitFor( () => {
+			expect( screen.getAllByText( 'Subscribe to all' ) ).toHaveLength( 2 );
+		} );
+	} );
+
+	it( 'shows "Unsubscribe from all" when at least one setting is enabled (WP)', async () => {
+		// Start with one WordPress.com settings enabled
+		const settingsWithWpcomEnabled: UserNotificationSettings = {
+			...defaultUserSettings,
+			wpcom: {
+				...defaultWpcomSettings,
+				marketing: true,
+			},
+		};
+
+		mockGetNotificationSettingsApi( settingsWithWpcomEnabled );
+
+		dashboardRender(
+			<>
+				<Snackbars />
+				<NotificationsExtras />
+			</>
+		);
+
+		await waitFor( () => {
+			// WordPress.com section should show "Unsubscribe from all" because all options are enabled
+			expect( screen.getByText( 'Unsubscribe from all' ) ).toBeVisible();
```

Comment:
Minor improvement, getByRole works better because we want to ensure it is a control instead of a random phrase on the page body.

Reply:
Indeed it would, updated. Thanks!